protocol/shannon/protocol.go

```diff
@@ -146,7 +159,7 @@ func (p *Protocol) BuildRequestContextForEndpoint(
 
 	// Retrieve the list of endpoints (i.e. backend service URLs by external operators)
 	// that can service RPC requests for the given service ID for the given apps.
-	endpoints, err := p.getAppsUniqueEndpoints(serviceID, permittedApps)
+	endpoints, err := p.getAppsUniqueEndpoints(ctx, serviceID, permittedApps)
 	if err != nil {
 		return nil, fmt.Errorf("BuildRequestContextForEndpoint: error getting endpoints for service %s: %w", serviceID, err)
 	}
```

Comment:
Could we log/debug the `selectedEndpoint` somewhere. It will be useful to follow where the request went when debugging.